docs/CNAME

```diff
@@ -0,0 +1 @@
+docs.openworm.org
```

Comment:
Just to check, this is needed only for mkdocs, right?

Reply:
yes, https://www.mkdocs.org/user-guide/deploying-your-docs/#custom-domains

We need `CNAME` in the root for github to know about the custom domain, and in the docs_dir for `mkdocs` to ensure it also gets put in the root of `gh-pages` when deployed.